### PR TITLE
partners: Add chroma partner package

### PR DESCRIPTION
**Pull request by akiradev0x**
**Description:** Adds chroma to the partners package. Tests & code mirror those in the community package.
**Dependencies:** None
**Twitter handle:** @akiradev0x

### CODE REVIEW

File: libs/partners/chroma/tests/persist_dir/chroma.sqlite3
Comment (efriis) [inline libs/partners/chroma/tests/persist_dir/chroma.sqlite3:1]: Is this necessary for testing, or can we generate this in a gitignored fashion during integration testing?

File: libs/partners/chroma/tests/unit_tests/test_vectorstores.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from langchain_chroma.vectorstores import Chroma
+from tests.integration_tests.fake_embeddings import (
+    FakeEmbeddings,
```
Comment (efriis) [inline libs/partners/chroma/tests/unit_tests/test_vectorstores.py:3]: import from core instead!

File: libs/partners/chroma/tests/integration_tests/test_vectorstores.py [hunk 2]
```diff
@@ -0,0 +1,457 @@
+"""Test Chroma functionality."""
+
+import uuid
+
+import pytest
+import requests
+from langchain_community.embeddings import FakeEmbeddings as Fak
```
Comment (efriis) [inline libs/partners/chroma/tests/integration_tests/test_vectorstores.py:7]: from core instead!

File: libs/partners/chroma/tests/integration_tests/fake_embeddings.py
Comment (efriis) [inline libs/partners/chroma/tests/integration_tests/fake_embeddings.py:1]: believe these are in core so can remove

File: libs/partners/chroma/langchain_chroma/vectorstores.py [hunk 3]
```diff
@@ -0,0 +1,798 @@
...
+    List,
+    Optional,
+    Tuple,
+    Type,
+    Union,
+)
+
+import numpy as np
+from chromadb import QueryResult
+from langchain_community.vectorstores.utils import maximal_marginal_relevance
```
Comment (efriis) [inline libs/partners/chroma/langchain_chroma/vectorstores.py:21]: don't rely on community - either import from core or copy function into package

File: libs/partners/chroma/langchain_chroma/vectorstores.py [hunk 4]
```diff
@@ -0,0 +1,798 @@
...
+        """Initialize with a Chroma client."""
+        try:
+            import chromadb
+            import chromadb.config
+        except ImportError:
+            raise ImportError(
+                "Could not import chromadb python package. "
+                "Please install it with `pip install chromadb`."
+            )
+
```
Comment (efriis) [inline libs/partners/chroma/langchain_chroma/vectorstores.py:89]: import at top instead

**Comment by efriis** [comment]
failing CI and would love to get ^ addressed to merge!

**Comment by killind-dev** [comment]
working on it

File: libs/partners/chroma/langchain_chroma/vectorstores.py (hunk 3 above)
Comment (killind-dev) [reply libs/partners/chroma/langchain_chroma/vectorstores.py]: No longer relying on community

File: libs/partners/chroma/tests/integration_tests/fake_embeddings.py
Comment (killind-dev) [reply libs/partners/chroma/tests/integration_tests/fake_embeddings.py:1]: For some reason the fake embeddings class from core causes all the retrieval integ tests to fail. The one in this file does not, it also matches the file in langchain community (it has its own fake_embeddings.py)

File: libs/partners/chroma/tests/integration_tests/test_vectorstores.py (hunk 2 above)
Comment (killind-dev) [reply libs/partners/chroma/tests/integration_tests/test_vectorstores.py]: fixed

File: libs/partners/chroma/tests/unit_tests/test_vectorstores.py (hunk 1 above)
Comment (killind-dev) [reply libs/partners/chroma/tests/unit_tests/test_vectorstores.py:2]: fixed

File: libs/partners/chroma/tests/persist_dir/chroma.sqlite3
Comment (killind-dev) [reply libs/partners/chroma/tests/persist_dir/chroma.sqlite3:1]: ignored

File: libs/partners/chroma/langchain_chroma/vectorstores.py [hunk 5]
```diff
@@ -0,0 +1,825 @@
...
+        This can be used to explicitly persist the data to disk.
+        It will also be called automatically when the object is destroyed.
+        """
+        if self._persist_directory is None:
+            raise ValueError(
+                "You must specify a persist_directory on"
+                "creation to persist the collection."
+            )
+
+        warnings.warn("chromadb^4 automatically persists when persist_directory is set")
```
Comment (efriis) [inline libs/partners/chroma/langchain_chroma/vectorstores.py:657]: @killind-dev should we just remove this method? It looked like it only had an implementation for an sdk version different than the required one for the partner package

Comment (atroyn) [reply libs/partners/chroma/langchain_chroma/vectorstores.py:657]: This is here for backward compatibility.

**Review by efriis — APPROVED** [review]
removed `def persist`

**Pull request merged**